modules/instance_template/variables.tf

```diff
@@ -153,6 +153,21 @@ variable "network_ip" {
   default     = ""
 }
 
+variable "additional_networks" {
+  description = "Additional network interface details for GCE, if any."
+  default     = []
+  type = list(object({
+    network            = string
+    subnetwork         = string
+    subnetwork_project = string
+    network_ip         = string
+    access_config = list(object({
```

Comment:
I'm not sure we should make this a list of objects—in practice, only one `access_config` is used. Can we uplevel `nat_ip` and `network_tier`?

Reply:
@morgante I understand your point and what saying is right, but I want to keep the access_config as a list for the following reasons

1. According to the [Compute Insert Spec](https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert#:~:text=string%2C%0A%20%20%20%20%20%20%22name%22%3A%20string%2C-,%22accessConfigs%22,-%3A%20%5B%0A%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22type%22%3A%20enum%2C%0A%20%20%20%20%20%20%20%20%20%20%22name), the access config is expected as a list
2. [Existing Access Config](https://github.com/terraform-google-modules/terraform-google-vm/blob/master/modules/instance_template/variables.tf#L217) data type is a list. I think there should be a consistency in the data structure in all places. Users should not be confused with a list data structure in one place and object data structure in another place